File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
bool alphanum(char c ){
    return(c>='A'&& c<='Z' || c >='a' && c<='z' || c >='0' && c<='9');
}
    bool isPalindrome(string s) {
        int n = s.length();
        int i = 0 ;
        int j = n-1;
        while( i < j ){
            while( i < j && !alphanum(s[i])){
                i++;
            }
            while( i < j && !alphanum(s[j])){
                j--;
            }
            if(tolower(s[i]) != tolower(s[j])){
                return false ; 
            }
            i++;
            j--;
        }
        return true ; 
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int i = 0 ;
        int j = s.length()-1 ; 

        while( i < j ){
            if(!Character.isLetterOrDigit(s.charAt(i))){
              i++;
              continue; 
            }
            if(!Character.isLetterOrDigit(s.charAt(j))){
                j--;
                continue;
            }
            char p = Character.toLowerCase(s.charAt(i)) ; 
             char q = Character.toLowerCase(s.charAt(j)) ; 

             if(p != q ) return false ; 
             i++;
             j--;
        }
        return true ; 
    }
}